examples/lib/stories/experimental/loading_screen.dart

```diff
@@ -0,0 +1,239 @@
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(message: 'onLoad started', progress: 1),
+    );
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(
+        message: 'Something happen while loading...',
+        progress: 50,
+      ),
+    );
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(
+        message: '...almost done!..',
+        progress: 90,
+      ),
+    );
+
+    add(
+      TextComponent(text: 'This is in-game text components')
+        ..anchor = Anchor.topCenter
+        ..x = size.x / 2
+        ..y = 50,
+    );
+
+    /// These components will receive notifications during game loop.
+    add(InGameProgressText(verticalPosition: 100, size: size));
+    add(InGameProgressText(verticalPosition: 150, size: size));
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    progressNotifier.reportLoadingProgress(
+      const ProgressMessage(
+        message: 'Finished! Be ready to play!',
+        progress: 100,
+      ),
+    );
+
+    await Future<void>.delayed(const Duration(seconds: 5));
+
+    /// You should to decide, what type of message would be trigger for
+    /// showing game widget. You should to sent such message manually and
+    /// also manually build game widget in loading screen component.
+    /// See [_LoadingScreenExampleWidgetState.build]
+    progressNotifier.reportLoadingProgress(const ProgressMessage.finished());
+
+    return super.onLoad();
+  }
+
+  /// This is just an example of sending messages to game components.
+  @override
+  void update(double dt) {
+    anyProgressEmulation += dt;
+    final probability = Random().nextInt(100);
+    if (probability < 1) {
+      messagesSent++;
+      progressNotifier.reportLoadingProgress(
+        ProgressMessage(
+          message: 'The notification message #$messagesSent',
+          progress: (anyProgressEmulation * 100).toInt(),
+        ),
+      );
+    }
+    super.update(dt);
+  }
+}
+
+/// A message strict type. You should to define an immutable message class for
+/// exchanging between game and components. You also could to use just [String]
+/// and even [dynamic], but the last option looks like heavy anti-pattern.
+@immutable
+class ProgressMessage {
+  const ProgressMessage({required this.message, required this.progress});
+
+  /// Just for fast illustration - a special constructor to indicate, that
+  /// [Game.onLoad] is finished.
+  const ProgressMessage.finished()
+      : message = '',
+        progress = 101;
+
+  final String message;
+  final int progress;
+}
+
+/// The loading screen widget with animated progress bar anf fade effect
+/// when displaying the game.
+class LoadingScreenExampleWidget extends StatefulWidget
+    with LoadingWidgetMixin<ProgressMessage> {
```

Comment:
Is there a special reason why we add the `LoadingWidgetMixin` here instead of the `State` class just below?
Usually I see mixins are added to State part of the `StatefulWidget`s next to `SingleTickerProviderStateMixin`

Reply:
The reason is an attempt to hide all stream setups from the end-user. 
I want to pass the already initialized `_LoadingProgressReceiver` to a user-defined widget, so I need to access this widget at build time. I haven't such access to the widget's state. Also, maybe your loading screen is just a stateless wrapper-widget? In this cause, the mixin will not work at all.

Reply:
Ah, got it, my trick does not work here. Thanks! 
Dont waste your time on reading my code now, please:  it has very critical problem!

Reply:
Done. 
Components with mixin are replaced by custom class with builder functions. 
Streams and futures are kept under hood. After loading is finished, unnecessary StreamBuilder is removed from widgets tree, so performance should not be affected. 

And yes, bugs seems to be fixed now =)